### DISCUSSION TITLE
problem when using svg

### DISCUSSION THREAD

**Discussion by marcoswebmasteroficial** [vercel/next.js#67756] [Help]
### Summary

I'm facing a problem when trying to use SVG in Next.js 14.2.4 using import Image from 'next/image';. When I try to render an SVG with the following code:`import Image from 'next/image';

```
function MyComponent() {
  return (
    <div>
      <Image
        src="/assets/icon.svg"
        alt="My SVG"
        width={200}
        height={200}
      />
    </div>
  );
}
```

**I'm getting the following error:`TypeError: Cannot read properties of undefined (reading 'call')**

I've already reviewed the Next.js documentation and tried adjusting the configuration in next.config.js using @svgr/webpack, but I still haven't been able to resolve the issue.

Any suggestions on how I can fix this error or an alternative approach to optimally display SVGs in Next.js 14?

Thank you in advance for your help!

**Comment by icyJoseph** [comment]
Works fine here: https://stackblitz.com/edit/nextjs-zbhmgm?file=app%2Fpage.tsx,app%2Flayout.tsx

Can you share more code, or specifics of your project? Maybe yo already fixed it?

**Reply by marcoswebmasteroficial** [reply]
It didn't work for me, https://github.com/marcoswebmasteroficial/boilerplate_nextjs/

**Reply by icyJoseph** [reply]
Err, should've posted the link to the repo earlier.

Without involving svgr, you can already do:

```tsx
'use client'
import { CardContainer, Title, Content, Text } from './style'

import Image from 'next/image'

interface CardProps {
  title: string
  description: string
  icon: string
}

const Card = ({ title, description, icon }: CardProps) => {
  return (
    <CardContainer>
      <div>
        <Title>{title}</Title>
        <Content>
          <Image
            src="/icons/styled_components.svg"
            alt="svg-next/image"
            width={100}
            height={90}
          />

          <Text>{description}</Text>
        </Content>
      </div>
    </CardContainer>
  )
}

export default Card


```

See, https://stackblitz.com/edit/github-2zjvgl?file=src%2Fcomponents%2FCard%2Findex.tsx


And if you want to skip using next/image, and have the SVG be used "like" a React component, follow these steps, https://react-svgr.com/docs/next/ - see it implemented here, https://stackblitz.com/edit/github-2zjvgl-3qxttn?file=src%2Fcomponents%2FCard%2Findex.tsx,next.config.mjs - Typically this one is more useful, because you can pass props (className for example) into the svg-react-component, to style it.

**Comment by marcoswebmasteroficial** [comment]
I was testing other things and still got the error
![Captura de tela 2024-07-15 194520](https://github.com/user-attachments/assets/9f43962b-1531-419f-850c-58737c7dba75)

**Reply by icyJoseph** [reply]
Make a branch with the code that produces that error, push it and share the link to the branch code

**Reply by icyJoseph** [reply]
The Stackblitz links here, https://github.com/vercel/next.js/discussions/67756#discussioncomment-10056012 both work, and they use your repository

**Reply by marcoswebmasteroficial** [reply]
I've tried everything, it doesn't work,stackblitz 
it runs normally apparently, but not on localhost

**Reply by icyJoseph** [reply]
I wonder if the PWA you are using is, partly responsible. Although, svgr should work regardless. I need to look at the code to see what's going on. Take a closer look at the Stackblitz - examples and the steps shown to setup svgr.

**Reply by marcoswebmasteroficial** [reply]
even if I remove the pwa it doesn't work,I looked at the svgr setup configuration, it doesn't work

**Reply by icyJoseph** [reply]
But what did you do... 

The Stackblitz link I shared, used svgr with your repository and it works.

**Reply by marcoswebmasteroficial** [reply]
As I said on Stackblitz, it doesn't work on localhost, download and test on your localhost

**Reply by icyJoseph** [reply]
It works locally too 😓 

I don't even know why I tried... it obviously had to work, both SVGR, and just using the path to the svg with next/image

**Reply by marcoswebmasteroficial** [reply]
Test on another machine, and it works normally, apparently an error is occurring in Windows 11, thank you, sorry for the frustration
